tests/thread_singleton_test.py

```diff
@@ -0,0 +1,73 @@
+import dace
+from dace import dtypes, nodes
+import numpy as np
+
+
+N = dace.symbol("N")
+
+
+@dace.program
+def cpu_singleton_program(A: dace.int32[N]):
+    for i in dace.map[0:N]:
+        A[i] = i
```

Comment:
With the CPU Singleton schedule it generates code like:
```cpp
#pragma omp parallel
{
    int i = omp_get_thread_num();
    {
        int __out;

        ///////////////////
        // Tasklet code (assign_13_8)
        __out = i;
        ///////////////////

        A[i] = __out;
    }
}
```

Since I am not that familiar with OpenMP, I'd have to study the docs to see if there is a better solution.